### ISSUE TITLE
Allow to fallback to VS Code marketplace for missing packages

### ISSUE THREAD

**Issue by Alikont** [coder/code-marketplace#52]
With a lot of package management feeds, it's common to support falling back to upstream feed for missing packages, so you can do all operations with a single feed url.

Service should return local packages if it's available, and fall back to real marketplace if it's not.

**Comment by code-asher** [comment]
I do not think we can specifically point to the VS Code marketplace for legal reasons.

But, it would be cool to implement a generic fallback so you can specify a different marketplace, like say Open VSX.

**Comment by code-asher** [comment]
Although I wonder if some kind of caching proxy is a better fit for this use case.

**Comment by Alikont** [comment]
What I want is to have a single URL that I can configure for people to have both private and public packages

For me it's enough to just proxy it

Yes, a generic "fallback url" is a good config, so it's users who bear legal responsibility

**Comment by garylavayou** [comment]
> Although I wonder if some kind of caching proxy is a better fit for this use case.

@code-asher Maybe some functionality like **Nexus repository manager**/**Docker registry** that can act as a proxy of an upstream extension registry (i.e., open-vsx.org).

**Comment by code-asher** [comment]
I also just learned about https://github.com/LOLINTERNETZ/vscodeoffline

Not an endorsement, I have never used it, but maybe it is worth looking at.
